consensus/parlia/parlia.go

```diff
@@ -537,6 +547,26 @@ func (p *Parlia) verifyHeader(chain consensus.ChainHeaderReader, header *types.H
 	if err := misc.VerifyForkHashes(chain.Config(), header, false); err != nil {
 		return err
 	}
+
+	parent, err := p.getParent(chain, header, parents)
+	if err != nil {
+		return err
+	}
+
+	// Verify the block's gas usage and (if applicable) verify the base fee.
+	if !chain.Config().IsLondon(header.Number) {
```

Comment:
It will be easier if we do it on London HF block, as this requires the least amount of code changes. Otherwise we would have to add more if-else branches which would complicate the code further.

Reply:
Ok, I don't know why that one was merged without merging this PR first. Shanghai upgrade builds on London upgrade. This means that the instruction sets will have to be updated again.